music/demos/transcription.ts

```diff
@@ -59,36 +91,57 @@ async function transcribe(oaf: mm.OnsetsAndFrames, batchLength: number) {
 
 async function transcribeFromAudio(oaf: mm.OnsetsAndFrames) {
   const audio = await loadBuffer(ORIGINAL_AUDIO_URL);
-  const expectedAudioNs: INoteSequence =
-      await fetch(`${MEL_CKPT_URL}/${EXPECTED_NS_SUFFIX}`)
-          .then((response) => response.json());
-  writeNoteSeqs('expected-audio-ns', [expectedAudioNs], undefined, true);
-
   const start = performance.now();
   const ns = await oaf.transcribeFromAudio(audio);
   writeTimer('audio-time', start);
   writeNoteSeqs('audio-results', [ns], undefined, true);
 
   document.getElementById('audio-match').innerHTML =
-      notesMatch(ns.notes, expectedAudioNs.notes) ?
+      notesMatch(ns.notes, expectedNs.notes) ?
       '<span style="color:green">TRUE</span>' :
       '<b><span style="color:red">FALSE</span></b>';
 }
 
-try {
+async function getAudioBufferFromBlob(blob: Blob) {
+  const fileReader = new FileReader();
+  return new Promise((resolve, reject) => {
+    fileReader.onerror = () => {
+      fileReader.abort();
+      reject(new DOMException('Something went wrong reading that file.'));
+    };
+    fileReader.onload = () => {
+      resolve(fileReader.result);
+    };
+    fileReader.readAsArrayBuffer(blob);
+  });
+}
+
+async function transcribeFromFile(blob: Blob) {
+  setLoadingMessage('file');
+  const audioCtx = new AudioContext();
+  const arrayBuffer: ArrayBuffer =
+      await getAudioBufferFromBlob(blob) as ArrayBuffer;
+  const audio: AudioBuffer = await audioCtx.decodeAudioData(arrayBuffer);
+
+  const audioEl = document.getElementById('filePlayer') as HTMLAudioElement;
+  audioEl.hidden = false;
+  audioEl.src = window.URL.createObjectURL(blob);
+
   const oafA = new mm.OnsetsAndFrames(AUD_CKPT_URL);
-  const oafM = new mm.OnsetsAndFrames(MEL_CKPT_URL);
+  const start = performance.now();
   oafA.initialize()
```

Comment:
Can we add something that signals to the users when the model is loading/loaded?

Reply:
we could, but we don't do this for any of the other demos rn.

Reply:
Please do since this one takes a while.